intg-denonavr/denon_remote.py

```diff
@@ -0,0 +1,1226 @@
+    avr.States.ON: ucapi.remote.States.ON,
+    avr.States.OFF: ucapi.remote.States.OFF,
+    avr.States.PAUSED: ucapi.remote.States.ON,
+    avr.States.PLAYING: ucapi.remote.States.ON,
+    avr.States.UNAVAILABLE: ucapi.remote.States.UNAVAILABLE,
+    avr.States.UNKNOWN: ucapi.remote.States.UNKNOWN,
+}
+
+
+# pylint: disable=R0903
+class DenonRemote(Remote):
+    """Representation of a Denon/Marantz AVR Remote entity."""
+
+    def __init__(self, device: AvrDevice, receiver: avr.DenonDevice, denon_media_player: DenonMediaPlayer):
+        """Initialize the class."""
+        self._device: avr.DenonDevice = receiver
+        self._denon_media_player: DenonMediaPlayer = denon_media_player
+        entity_id = create_entity_id(receiver.id, EntityTypes.REMOTE)
+        features = [Features.SEND_CMD, Features.ON_OFF, Features.TOGGLE]
+        super().__init__(
+            entity_id,
+            f"{device.name} Remote",
+            features,
+            attributes={
+                Attributes.STATE: receiver.state,
+            },
+            simple_commands=simplecommand.get_simple_commands(device),
+            button_mapping=REMOTE_BUTTONS_MAPPING,
+            ui_pages=DenonRemote._get_remote_ui_pages(),
+        )
+
+    async def command(self, cmd_id: str, params: dict[str, Any] | None = None) -> StatusCodes:
+        """
+        Remote entity command handler.
+
+        Called by the integration-API if a command is sent to a configured remote entity.
+
+        :param cmd_id: command
+        :param params: optional command parameters
+        :return: status code of the command request
+        """
+        return await self._handle_command(cmd_id, params)
+
+    @staticmethod
+    def state_from_avr(avr_state: avr.States) -> ucapi.remote.States:
+        """
+        Convert AVR state to UC API remote state.
+
+        :param avr_state: Denon/Marantz AVR state
+        :return: UC API remote state
+        """
+        if avr_state in REMOTE_STATE_MAPPING:
+            return REMOTE_STATE_MAPPING[avr_state]
+        return ucapi.remote.States.UNKNOWN
+
+    def filter_changed_attributes(self, update: dict[str, Any]) -> dict[str, Any]:
+        """
+        Filter the given attributes and return only the changed values.
+
+        :param update: dictionary with attributes.
+        :return: filtered entity attributes containing changed attributes only.
+        """
+        attributes = {}
+
+        if Attributes.STATE in update:
+            state = DenonRemote.state_from_avr(update[Attributes.STATE])
+            attributes = helpers.key_update_helper(Attributes.STATE, state, attributes, self.attributes)
+
+        return attributes
+
+    async def _handle_command(self, cmd_id: str, params: dict[str, Any] | None = None) -> StatusCodes:
+        command = ""
+        if params:
+            command = params.get("command", "")
+
+        if command == "":
+            command = f"remote.{cmd_id}"
+
+        match command:
```

Comment:
This could be simplified by directly matching `cmd_id`. Or is there a reason for this indirection?
From the remote-entity specification:
> - A command name may not be any of the defined `cmd_id` commands: on, off, toggle, send_cmd, send_cmd_sequence

Reply:
No reason, I used the Yamaha driver for inspiration. I'll remove it

Reply:
Added a workaround